lib/hub_connection.dart

```diff
@@ -15,35 +18,56 @@ const int DEFAULT_PING_INTERVAL_IN_MS = 15 * 1000;
 enum HubConnectionState {
   /// The hub connection is disconnected.
   Disconnected,
-
+  /// The hub connection is connecting.
+  Connecting,
   /// The hub connection is connected.
   Connected,
+  /// The hub connection is disconnecting.
+  Disconnecting,
+  /// The hub connection is reconnecting.
+  Reconnecting,
 }
 
-typedef InvocationEventCallback = void Function(
-    HubMessageBase invocationEvent, Exception error);
-typedef MethodInvacationFunc = void Function(List<Object> arguments);
-typedef ClosedCallback = void Function(Exception error);
+typedef InvocationEventCallback = void Function(HubMessageBase invocationEvent, Exception error);
+typedef MethodInvocationFunc = void Function(List<Object> arguments);
+typedef ClosedCallback = void Function({Exception error});
```

Comment:
{Exception error} in ClosedCallback does not really need to be optional please revert it back as current consumers need less changes in their codes after upgrade packages.

Reply:
The change to optional stems from the basis that the PR is to bring the codebase into line with the aspnetcore 3.1 Typescript client which shows onclose and onreconnecting callbacks as having optional Error parameters to their callbacks.  I want to keep the API as close to the TS client as possible.